src/components/PortfolioPage.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { graphql } from 'gatsby';
+import { MDXProvider } from '@mdx-js/react';
+import { MDXRenderer } from 'gatsby-plugin-mdx';
+import { Link } from 'gatsby';
+
+const shortcodes = { Link }; // Provide common components here
+
+export default function PortfolioPage({ data: { mdx } }) {
```

Comment:
I'm using `.js` instead of `.tsx` for the React files, coz I'm a bit slow when working with the `tsx` files. I'll still use `ts` for the backend stuff.

Reply:
I'll take a stab at fixing all the typing issues

Reply:
ah, there is no typing issue here, I'm just planning to use .js here so that I won't face any issues later when I add more complicated stuff

Reply:
We should try keep everything typed, if you face any typing issues, its fine to leave them since gatsby will still build :)
Then we can try fix them in the future